encoding/streams/decode-utf8.any.js

```diff
@@ -40,22 +40,25 @@ promise_test(async () => {
                       'the output should be in one chunk');
 }, 'a trailing empty chunk should be ignored');
 
-promise_test(async () => {
-  const buffer = new ArrayBuffer(3);
-  const view = new Uint8Array(buffer, 1, 1);
-  view[0] = 65;
-  new MessageChannel().port1.postMessage(buffer, [buffer]);
-  const input = readableStreamFromArray([view]);
-  const output = input.pipeThrough(new TextDecoderStream());
-  const array = await readableStreamToArray(output);
-  assert_array_equals(array, [], 'no chunks should be output');
-}, 'decoding a transferred Uint8Array chunk should give no output');
-
-promise_test(async () => {
-  const buffer = new ArrayBuffer(1);
-  new MessageChannel().port1.postMessage(buffer, [buffer]);
-  const input = readableStreamFromArray([buffer]);
-  const output = input.pipeThrough(new TextDecoderStream());
-  const array = await readableStreamToArray(output);
-  assert_array_equals(array, [], 'no chunks should be output');
-}, 'decoding a transferred ArrayBuffer chunk should give no output');
+[ArrayBuffer, SharedArrayBuffer].forEach((ArrayBufferOrSharedArrayBuffer) => {
+  promise_test(async () => {
+    const buffer = new ArrayBufferOrSharedArrayBuffer(3);
+    const view = new Uint8Array(buffer, 1, 1);
+    view[0] = 65;
+    new MessageChannel().port1.postMessage(buffer, [buffer]);
```

Comment:
This test cannot be generalized this easily unfortunately. We'd also have to add headers and then we can no longer use `.any.js` syntax. I'm also not really sure if streams work with `SharedArrayBuffer`. @ricea?

Reply:
Streams are agnostic about SharedArrayBuffer[1]. Up until now I have interpreted

> Values of the IDL buffer source types are represented by objects of the corresponding ECMAScript class, with the additional restriction that unless the type is associated with the [AllowShared] extended attribute, they can only be backed by ECMAScript ArrayBuffer objects, and not SharedArrayBuffer objects.

from the WebIDL standard to imply that

> Let bufferSource be the result of converting chunk to a BufferSource.

in the encoding standard means that we don't accept SharedArrayBuffers.

---

[1] Except for byte streams. But this standard doesn't currently use byte streams.

Reply:
I see, I guess we want to track addressing that separately or should we add SharedArrayBuffer support there at the same time?

Reply:
Either is fine with me. The changes should not be large. If we do it in this PR then we also need to change https://github.com/web-platform-tests/wpt/blob/master/encoding/streams/decode-bad-chunks.any.js and https://github.com/web-platform-tests/wpt/blob/master/encoding/streams/encode-bad-chunks.any.js which explicitly assert that SharedArrayBuffer doesn't work.

Reply:
@Bnaya are you interested in taking these on as well?

Reply:
If i got it right:
need to remove the fail-test with SAB from here:
https://github.com/web-platform-tests/wpt/blob/master/encoding/streams/decode-bad-chunks.any.js

Not sure what do to in `encode-bad-chunks.any.js`

Reply:
> If i got it right:
need to remove the fail-test with SAB from here:
https://github.com/web-platform-tests/wpt/blob/master/encoding/streams/decode-bad-chunks.any.js

Correct.

> Not sure what do to in `encode-bad-chunks.any.js`

Sorry, `encode-bad-chunks.any.js` doesn't need anything doing. My mistake.

Reply:
I think we might also want to branch `decode-utf8.any.js` as you did before (although I can't tell exactly as that's overwritten now), but not the last two tests. That way we know for sure that `TextDecoderStream` also accepts `SharedArrayBuffer`.

Reply:
Done, see the last commit